src/components/Amalgamation/AmalgamatingBusinesses.vue

```diff
@@ -0,0 +1,250 @@
+
+  readonly BusinessLookupServices = BusinessLookupServices
+
+  // Cancel button in "Add an Amalgamating Business" is pressed.
+  addAmalgamatingBusinessCancel (): void {
+    this.isAddingAmalgamatingBusiness = false
+    // Enable buttons
+    this.addAmalgatingBusinessDisabled = false
+    this.addAmalgatingForeignBusinessDisabled = false
+  }
+
+  // "Add an Amalgamating Business" button is pressed.
+  onAddBusinessClick (): void {
+    this.isAddingAmalgamatingBusiness = true
+    this.isAddingAmalgamatingForeignBusiness = false
+    // Disable buttons
+    this.addAmalgatingBusinessDisabled = true
+    this.addAmalgatingForeignBusinessDisabled = true
+  }
+
+  // "Add an Amalgamating Foreign Business" button is pressed.
+  onAddForeignBusinessClick (): void {
+    this.isAddingAmalgamatingBusiness = false
+    this.isAddingAmalgamatingForeignBusiness = true
+    // Disable buttons (Please comment out the lines below or remove in part 2)
+    // this.addAmalgatingBusinessDisabled = true
+    // this.addAmalgatingForeignBusinessDisabled = true
+  }
+
+  async saveAmalgamatingBusiness (businessLookup: BusinessLookupIF): Promise<void> {
+    // Get the amalgamating business information
+    // Will have a different format depending on the business
+    let business = await LegalServices.fetchBusinessInfo(businessLookup.identifier)
+      .then((response) => {
+        return response?.data?.business
+      }).catch(() => {
+        return businessLookup
+      })
+
+    // Get the address of the amalgamating business
+    if (businessLookup.identifier && business.foundingDate) {
+      const addresses = await LegalServices.fetchAddresses(businessLookup.identifier)
+        .then((data) => {
+          // SP and GP have businessOffice instead of registeredOffice
+          return data?.registeredOffice || data?.businessOffice
+        }).catch(() => {
+          return undefined
+        })
+      if (addresses) {
+        business.officeAddress = addresses
+      }
+    }
+
+    // If the amalgamating businesses array is not empty, check if identifier already exists.
+    // If identifier already exists, don't add the business to the array.
+    if (this.amalgamatingBuinesses.length > 0) {
+      const businessExists = this.amalgamatingBuinesses.find(function (id) {
+        return id.identifier === business.identifier
+      })
+      if (!businessExists) this.amalgamatingBuinesses.push(business)
+    } else {
+      this.amalgamatingBuinesses.push(business)
+    }
+
+    // Set the amalgamated businesses array in the store.
+    this.setAmalgamatingBusinesses(this.amalgamatingBuinesses)
+    // Close the "Add an Amalgamating Business" Panel.
+    this.addAmalgamatingBusinessCancel()
+  }
+}
+</script>
+
+<style lang="scss" scoped>
+@import '@/assets/styles/theme.scss';
+
+.v-btn:not(.v-btn--round).v-size--default {
+  height: 44px;
+}
+
+// Overriding the section container class right padding.
+.section-container {
+  padding-right: 0rem;
```

Comment:
`pr-0` doesn't work on the v-card element?

Reply:
It does actually but it looked a bit better doing this (a bit less spacing). IDK, maybe I'm imagining things haha. Will go for pr-0.

Reply:
Yep, you're absolutely right. Will definitely be removing this!

Reply:
(Yes, in some cases I wish the linter would allow us to have multiple things on the same line, but alas.)

For example, `<p id="boo" class="font-weight-bold:">BOO!</p>` is nice and simple, but lint wants it on 6 lines!

Reply:
Lint is a double-edged sword isn't it 😆